_posts/2015-09-26-using-vim-to-drive-tddd.md

```diff
@@ -0,0 +1,98 @@
+
+## Specs Should Run in Vim
+
+In order to actually follow TDD I need to be running my tests constantly. If I want to run my tests constantly I need it to be easy -  preferably not requiring any sort of context switch. Running specs in Vim would help me to avoid the context switch but could it be easy?
+
+Thanks to the thoughtbot team and their Vim plugin, [vim-rspec](https://github.com/thoughtbot/vim-rspec), it can be!
+
+This plugin adds functions for running all specs, the entire current spec file, the nearest spec(s), and to re-run the most recent spec. These functions can be [mapped to leader commands](https://github.com/thoughtbot/vim-rspec#key-mappings) to allow you to run them quickly.
+
+<img src='/images/run_spec_default.gif'></img>
+
+To really get into the TDD flow I run specs basically anytime I am in normal mode for more than a few seconds, with this plugin was essentially spamming my new leader commands. I find that unless I am quickly editing a recent change I generally "sit" in normal mode while I am thinking of my next move. This thinking time lends it self to running specs.
+
+While I am sure most developers trying to show off the benefits of TDD don't actually run the specs as often as they do for demos I have found there is some small pleasure in this habit. I will run specs even when I know what the failure is going to be (there is a certain pride in knowing what is going to cause the failure). However, when you do this too much you do run the risk of actually slowing down development.  I was no longer sitting in Normal mode thinking up solutions, instead I was staring at RSpec running, losing context of where I was in my code. I wanted to be able to run my specs **and** still browse my code.
+
+This brings me to my next level of Vim TDD - running specs in a new window within the current buffer.
+
+## Specs Shouldn't Stop Development
+
+To be fair the realization that I didn't have to lose focus on Vim while running specs also came from thoughtbot. While watching a [talk by Ben Orenstein](https://www.youtube.com/watch?v=PU3qIVAO9aM) he pointed out that when he ran specs the output came up in a little pop-up Vim window **and** if they all passed the window would disappear.
+
+I knew what I wanted to do was possible, now all I had to do was figure out how it was done.
+
+Luckily Ben Orenstein is a generous guy and [shares his dotfiles](https://github.com/r00k/dotfiles) and even has a [nice comment](https://github.com/r00k/dotfiles/blob/master/vimrc#L280-L282) letting me know where the test running stuff is.
+
+From the`vimrc` of Ben Orenstein:
```

Comment:
Originally may have intentionally tried to go the phrasing but kinda awkward. Not sure if saying something old "from Ben's vimrc" would be too informal.